daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-cloudevents.md

```diff
@@ -55,20 +63,115 @@ As another example of a v1.0 CloudEvent, the following shows data as XML content
 
 ```json
 {
-    "specversion" : "1.0",
-    "type" : "xml.message",
-    "source" : "https://example.com/message",
-    "subject" : "Test XML Message",
-    "id" : "id-1234-5678-9101",
-    "time" : "2020-09-23T06:23:21Z",
-    "datacontenttype" : "text/xml",
-    "data" : "<note><to>User1</to><from>user2</from><message>hi</message></note>"
+  "topic": "orders",
+  "pubsubname": "order_pub_sub",
+  "traceid": "00-113ad9c4e42b27583ae98ba698d54255-e3743e35ff56f219-01",
+  "tracestate": "",
+  "data" : "<note><to></to><from>user2</from><message>Order</message></note>",
+  "id" : "id-1234-5678-9101",
+  "specversion" : "1.0",
+  "datacontenttype" : "text/xml",
+  "subject" : "Test XML Message",
+  "source" : "https://example.com/message",
+  "type" : "xml.message",
+   "time" : "2020-09-23T06:23:21Z"
+}
+```
+
+## Replace Dapr generated CloudEvents values
+
+Dapr automatically generates several CloudEvent properties. You can replace these generated CloudEvent properties by providing the following optional metadata key/value:
+
+- `cloudevent-id`: overrides `id`
+- `cloudevent-source`: overrides `source`
+- `cloudevent-type`: overrides `type`
+- `cloudevent-traceid`: overrides `traceid`
+- `cloudevent-tracestate`: overrides `tracestate`
+- `cloudevent-traceparent`: overrides `traceparent`
+
+The ability to replace CloudEvents properties using these metadata properties applies to all pub/sub components.
+
+### Example
+
+For example, to replace the `source` and `id` values from [the CloudEvent example above]({{< ref "#cloudevents-example" >}}) in code:
+
+{{< tabs "Python" ".NET" >}}
+ <!-- Python -->
+{{% codetab %}}
+
+```python
+with DaprClient() as client:
+    for i in range(1):
+        order = {'orderId': i}
+        # Publish an event/message using Dapr PubSub
+        result = client.publish_event(
+            pubsub_name='order_pub_sub',
+            topic_name='orders',
+            publish_metadata={'cloudevent-id: 'd99b228f-6c73-4e78-8c4d-3f80a043d317', cloudevent-source: 'payment'}
+        )
+```
+
+{{% /codetab %}}
+
+ <!-- .NET -->
+{{% codetab %}}
+
+```csharp
+for (int i = 1) {
```

Comment:
I meant removing the for loop entirely. Like this

```
var order = new Order("order1");
using var client = new DaprClientBuilder().Build();

// Override cloudevent metadata
var metadata = new Dictionary<string,string>() {
       { "cloudevent.source", "payment" },
       { "cloudevent.id", "d99b228f-6c73-4e78-8c4d-3f80a043d317" }
}

// Publish an event/message using Dapr PubSub
await client.PublishEventAsync("order_pub_sub", "orders", order, metadata);
Console.WriteLine("Published data: " + order);

await Task.Delay(TimeSpan.FromSeconds(1));
```

Reply:
ohh ok I misunderstood - fixing.